clang/lib/Interpreter/IncrementalParser.cpp

```diff
@@ -413,7 +413,9 @@ void IncrementalParser::CleanUpPTU(PartialTranslationUnit &PTU) {
     if (!ND)
       continue;
     // Check if we need to clean up the IdResolver chain.
-    if (ND->getDeclName().getFETokenInfo())
+    if (ND->getDeclName().getFETokenInfo() &&
+        CI->getPreprocessor().isIncrementalProcessingEnabled() &&
```

Comment:
Here the incremental processing is guaranteed. We do not need to check.

Reply:
good point. Removed.